lib/modules/datasource/maven/util.ts

```diff
@@ -128,6 +204,58 @@ export async function checkHttpResource(
   }
 }
 
+async function checkS3Resource(
+  pkgUrl: url.URL | string
+): Promise<HttpResourceCheckResult> {
+  try {
+    const s3Url = parseS3Url(pkgUrl.toString());
+    const response = await getS3Client().headObject(s3Url);
+    // istanbul ignore next
+    if (response.DeleteMarker) {
+      return 'not-found';
+    }
+    if (response.LastModified) {
+      return response.LastModified;
+    }
+    return 'found';
+  } catch (err) {
+    if (isS3NotFound(err)) {
+      return 'not-found';
+    } else {
+      // istanbul ignore next
+      logger.debug(
+        { pkgUrl, name: err.name, message: err.message },
+        `Can't check S3 resource existence`
+      );
+    }
+    // istanbul ignore next
+    return 'error';
+  }
+}
+
+export async function checkResource(
+  http: Http,
+  pkgUrl: url.URL | string
+): Promise<HttpResourceCheckResult> {
+  const parsedUrl =
+    typeof pkgUrl === 'string' ? (parseUrl(pkgUrl) as url.URL) : pkgUrl;
```

Comment:
don't cast better create a refactor PR first to use non deprecated global `URL`  class and our url util.

Reply:
You want me to refactor the use of `URL` _before_ being able to merge this PR?

Reply:
Please see https://github.com/renovatebot/renovate/pull/15109